entrypoint.sh

```diff
@@ -21,9 +21,85 @@ if [ "$AVD" ]; then
     socat tcp-listen:5555,bind=127.0.0.1,fork tcp:$AVD & SOCAT_PID=$!
 fi
 
-start
+swarmUrl="$JENKINS_MASTER_URL/swarm/swarm-client.jar"
+swarmPath="/swarm-client.jar"
 
-if [ "$AVD_PID" ]; then
-    kill $SOCAT_PID
-fi
+startTime=$(date +%s)
+period=10
+success=1
+
+attemptTimeout() {
```

Comment:
moving into separate method seems useless especially taking into account that you use hardcoded $period value...
if you want to preserve this log echo message I'd recommend to use local time parameter:
`local timeout=$1`

and execute from the code
`attemptTimeout $period`